src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java

```diff
@@ -73,38 +78,29 @@ public void drive(ChassisSpeeds speeds) {
 
 	@Override
 	public void periodic() {
+
+		double[] states = new double[8];
+		for (int i = 0; i < 4; i++) states[i * 2 + 1] = modules[i].getTargetState().speedMetersPerSecond;
+		for (int i = 0; i < 4; i++)
+			states[i * 2] = modules[i].getTargetState().angle.getRadians();
+		Logger.recordOutput("Target States", states);
+		for (int i = 0; i < 4; i++) states[i * 2 +1] = modules[i].getMeasuredState().speedMetersPerSecond;
+		for (int i = 0; i < 4; i++)
+			states[i * 2] = modules[i].getMeasuredState().angle.getRadians();
+		Logger.recordOutput("Measured States", states);
+
+
 		//if statment is so that the telop wont run if selfdrive is on.
-			for (SwerveModule module : modules) {
-				module.teleop();
-			}
-		
-			pose_est.update(imu.yaw(),getPositions());
-			LimelightHelpers.Results result = LimelightHelpers.getLatestResults("limelight").targetingResults;
-			
-			if(LimelightHelpers.getTV("limelight")){
-
-				//System.out.println(result.getBotPose2d());
-				double tl = result.latency_pipeline;
-				double cl = result.latency_capture;
-				
-				double timeStamp = Timer.getFPGATimestamp() - (tl/1000.0) - (cl/1000.0);
-
-
-				pose_est.addVisionMeasurement(result.getBotPose2d(), timeStamp);
-
-			}
-			field.setRobotPose(pose_est.getEstimatedPosition());
-			
-			double[] curr_pos = {
-                pose_est.getEstimatedPosition().getX(),
-                pose_est.getEstimatedPosition().getY(),
-                pose_est.getEstimatedPosition().getRotation().getRadians()
-        };
-		currentPos.setDoubleArray(curr_pos);
-		
+		for (SwerveModule module : modules) {
+			module.teleop();
+		}
+
+		speedMultiplier = speedMultiplierWidget.getEntry().get().getDouble();
```

Comment:
Same are above this code should be in the input Manager

Reply:
@GoofyCoderGuy @AreaProgram Can you guys work this into the input manager

Reply:
The input manager doesn't have a periodic function since it is a regular class and the shuffleboard values require that to work properly.

Reply:
Yes, that slider should be receiving constant data from shuffleboard to manage the speed of the robot. I do not remember working on that slider as the sliders I made were commented out because they were not working.

Reply:
InputManager extends SubsystemBase, base on the docs it should have periodic (https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/SubsystemBase.html)